controllers/synapse/mautrixsignal/signald_deployment.go

```diff
@@ -66,44 +64,25 @@ func (r *MautrixSignalReconciler) reconcileSignaldDeployment(ctx context.Context
 }
 
 // deploymentForSynapse returns a synapse Deployment object
-func (r *MautrixSignalReconciler) deploymentForSignald(ms *synapsev1alpha1.MautrixSignal, objectMeta metav1.ObjectMeta) (*appsv1.Deployment, error) {
-	ls := labelsForSignald(ms.Name)
-	replicas := int32(1)
-	signaldPVCName := objectMeta.Name
-
-	dep := &appsv1.Deployment{
-		ObjectMeta: objectMeta,
-		Spec: appsv1.DeploymentSpec{
-			Replicas: &replicas,
-			Selector: &metav1.LabelSelector{
-				MatchLabels: ls,
-			},
-			Template: corev1.PodTemplateSpec{
-				ObjectMeta: metav1.ObjectMeta{
-					Labels: ls,
-				},
-				Spec: corev1.PodSpec{
-					Containers: []corev1.Container{{
-						Image: "docker.io/signald/signald:0.23.0",
-						Name:  "signald",
-						VolumeMounts: []corev1.VolumeMount{{
-							Name:      "signald",
-							MountPath: "/signald",
-						}},
-					}},
-					Volumes: []corev1.Volume{{
-						Name: "signald",
-						VolumeSource: corev1.VolumeSource{
-							PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
-								ClaimName: signaldPVCName,
-							},
-						},
-					}},
-				},
-			},
-		},
+func (r *MautrixSignalReconciler) deploymentForSignald(ms *synapsev1alpha1.MautrixSignal) (*appsv1.Deployment, error) {
+	type deploymentExtraValues struct {
+		synapsev1alpha1.MautrixSignal
+		Labels map[string]string
+		Name   string
+	}
+
+	extraValues := deploymentExtraValues{
+		MautrixSignal: *ms,
+		Labels:        labelsForSignald(ms.Name),
+		Name:          GetSignaldResourceName(*ms),
```

Comment:
Why not just do this in the template itself?

Reply:
I'd argue it's best to pass those values for 2 reasons:
1. It paves the way for a potential future generic deployment templates.
2. It provides consistency for labels and name across the `Signald` resources.

Reply:
Fair enough. Does GetSignaldResourceName need to be exported? Is it used outside of this package? If so, should it live in a standalone package? (Not utils :D)